### PR TITLE
Audit change requests

### PR DESCRIPTION
**Pull request by Rhian**
audit

### Description of change

This PR adds entries to the audit log every time a change request is created or updated.

### Story Link

https://trello.com/c/EhLn4tmq/302-create-audit-logs-for-change-requests

### CODE REVIEW

File: app/models/description_change_request.rb [hunk 1]
```diff
@@ -21,4 +23,9 @@ def rejected_reason_is_present?
   def set_previous_application_description
     self.previous_description = planning_application.description
   end
+
+  def set_sequence
+    change_requests = PlanningApplication.find(planning_application.id).description_change_requests
+    increment_sequence(change_requests)
+  end
```
Comment (james) [inline app/models/description_change_request.rb:30]: Could this repeated method not be pushed up into the parent ChangeRequest model?

Comment (james) [reply app/models/description_change_request.rb:30]: oh, I see, it's a different type each time. 🤔

Comment (james) [reply app/models/description_change_request.rb:30]: Would:
```
  def set_sequence
    self.sequence = self.where(planning_application: planning_application).count + 1
  end
```

work in ChangeRequest for all types?

File: app/views/audits/index.html.erb [hunk 2]
```diff
@@ -21,13 +21,17 @@
...
           <td class="govuk-table__cell"> User deleted </td>
         <% end %>
         <td colspan="2" class="govuk-table__cell">
-          <%= activity(audit_item.activity_type, audit_item.activity_information) %>
+            <%= activity(audit_item.activity_type, audit_item.activity_information) %>
+          <% if audit_item.activity_type.include?("change") %>
+              <%= simple_format(audit_item.audit_comment, wrapper_tag: "div",) if audit_item.audit_comment.present? %>
+          <% else %>
             <%= simple_format(audit_item.audit_comment, { class: "audit_details" }, wrapper_tag: "div",) if audit_item.audit_comment.present? %>
+          <% end %>
```
Comment (james) [inline app/views/audits/index.html.erb:34]: I'm thinking this is getting quite complicated, and now might be a good time for a refactor of how we do this:

I was thinking the dumber (simpler) option is to do something like:

```
<% @audits.each do |audit_item| %>
   <%= render partial: "audit_logs/types/#{audit_item. activity_type}" %>
<% end %>
```

and then create a partial for each activity type row. What do you think about that?

File: app/helpers/audit_helper.rb [hunk 3]
```diff
@@ -59,6 +59,6 @@ def activity(type_of_activity, args = nil)
 
   def define_api_user(audit)
     change_requests = %w[description_change_request_received red_line_boundary_change_request_received document_change_request_received document_create_request_received]
-    change_requests.include?(audit.activity_type) ? "Applicant / Agemt" : audit.api_user.name
+    change_requests.include?(audit.activity_type) ? "Applicant / Agent via #{audit.api_user.name}" : audit.api_user.name
```
Comment (james) [inline app/helpers/audit_helper.rb:62]: I think we want to say "Applicant / Agent via #{audit.api_user.name}" for all api actions (I mean the only other one is the creation of an application, which is also the applicant via the API).

File: app/views/audits/index.html.erb [hunk 4]
```diff
@@ -21,16 +21,18 @@
...
         <% elsif audit_item.api_user %>
-          <td class="govuk-table__cell"><%= define_api_user(audit_item) %></td>
+          <td class="govuk-table__cell"><%= audit_item.api_user.name %></td>
+        <% elsif audit_item.activity_type.include?("received") %>
+          <td class="govuk-table__cell"> Agent / Applicant via API </td>
         <% else %>
           <td class="govuk-table__cell"> User deleted </td>
         <% end %>
         <td colspan="2" class="govuk-table__cell">
             <%= activity(audit_item.activity_type, audit_item.activity_information) %>
```
Comment (james) [inline app/views/audits/index.html.erb:31]: Could we also move this into the partials and remove this `activity` method?

**Review by james — APPROVED** [review]

File: app/helpers/audit_helper.rb (hunk 3 above)
Comment (Rhian) [reply app/helpers/audit_helper.rb:62]: Unfortunately at the moment we don't submit the api_user from `bops-applicants` so it's a very temporary fix until we can deploy the permanent fix to `bops_applicants`

**Pull request merged**